src/core/pyproxy.js

```diff
@@ -214,10 +212,26 @@ Module.pyproxy_destroy = function (proxy, destroyed_msg) {
   }
   let ptrobj = _getPtr(proxy);
   Module.finalizationRegistry.unregister(proxy);
+  destroyed_msg ||= "Object has already been destroyed";
+  let proxy_type = proxy.type;
+  let proxy_repr;
+  try {
+    proxy_repr = proxy.toString();
+  } catch (e) {
+    if (e.pyodide_fatal_error) {
+      throw e;
+    }
+  }
   // Maybe the destructor will call JavaScript code that will somehow try
   // to use this proxy. Mark it deleted before decrementing reference count
   // just in case!
   proxy.$$.ptr = null;
+  destroyed_msg += "\n" + `The object was of type ${proxy_type} and `;
```

Comment:
```suggestion
  destroyed_msg += "\n" + `The object was of type '${proxy_type}' and `;
```

Reply:
I agree with quotes around the type. My worry about quotes around the repr is that many reprs themselves contain quotes. Not sure how much that matters... Note also that the repr of a string will be surrounded by single quotes unless the string contains a single quote and does not contain a double quote. So surrounding with double quotes is probably preferable.